### PR TITLE
documentation(EJ2-861557):Need to update the documentation for the virtualization in EJ2 dropdown components

### PR DESCRIPTION
**Pull request by mohanrajsf3477**
Need to update the documentation for the virtualization in EJ2 dropdown components

### CODE REVIEW

File: ej2-javascript/auto-complete/virtual-scroll.md [hunk 1]
```diff
@@ -17,6 +17,15 @@ During virtual scrolling, the data retrieved from the data source depends on the
  
 When fetching data from the data source, the [actionBegin](../api/auto-complete/#actionbegin) event is triggered before data retrieval begins. Then, the [actionComplete](../api/auto-complete/#actioncomplete) event is triggered once the data is successfully fetched.
 
+Please note that when the enableVirtualization property is enabled, the 'take' property of the Query parameter may not function as expected, as it is internally managed within our source code.
```
Comment (PrinceOliver) [inline ej2-javascript/auto-complete/virtual-scroll.md:20]: Please note that when the enableVirtualization property is enabled, the Skip and 'take' property of the Query parameter cannot be modified in the user end, as it is internally managed within our source code based on the popup height.

File: ej2-javascript/auto-complete/virtual-scroll.md [hunk 2]
```diff
@@ -17,6 +17,15 @@ During virtual scrolling, the data retrieved from the data source depends on the
...
+
+
+## Binding local data
+
+The AutoComplete can generate its list items through an array of complex data. For this, the appropriate columns should be mapped to the [fields](../api/drop-down-list/#fields) property.
```
Comment (PrinceOliver) [inline ej2-javascript/auto-complete/virtual-scroll.md:25]: Can you add explanation how virtualization is handled in local data ? include details like whether actionBegin and actionComplete will be triggered.

File: ej2-javascript/auto-complete/virtual-scroll.md [hunk 3]
```diff
@@ -44,7 +53,70 @@ When fetching data from the data source, the [actionBegin](../api/auto-complete/
...
+{% include code-snippet/autocomplete/virtual-scroll-remote/index.js %}
+{% endhighlight %}
+{% highlight html tabtitle="index.html" %}
+{% include code-snippet/autocomplete/virtual-scroll-remote/index.html %}
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "page.domainurl/code-snippet/autocomplete/virtual-scroll-remote" %}
+{% endif %}
+
```
Comment (PrinceOliver) [inline ej2-javascript/auto-complete/virtual-scroll.md:89]: Include sections for filtering and incremental search

File: ej2-javascript/combo-box/virtual-scroll.md [hunk 4]
```diff
@@ -17,6 +17,17 @@ During virtual scrolling, the data retrieved from the data source depends on the
  
 When fetching data from the data source, the [actionBegin](../api/combo-box/#actionbegin) event is triggered before data retrieval begins. Then, the [actionComplete](../api/combo-box/#actioncomplete) event is triggered once the data is successfully fetched.
 
+Furthermore, IncrementalSearch is supported with virtualization in the Combobox component. When a key is typed, the focus is moved to the respective element in the open popup state. In the closed popup state, the focus shifts to the corresponding item based on the typed key.
```
Comment (PrinceOliver) [inline ej2-javascript/combo-box/virtual-scroll.md:20]: Add separate section for incremental search and filtering

File: ej2-javascript/drop-down-list/virtual-scroll.md [hunk 5]
```diff
@@ -13,10 +13,22 @@ domainurl: ##DomainURL##
 
 Dropdown list virtualization is a technique used to efficiently render extensive lists of items while minimizing the impact on performance. This method is particularly advantageous when dealing with large datasets because it ensures that only a fixed number of DOM (Document Object Model) elements are created. When scrolling through the list, existing DOM elements are reused to display relevant data instead of generating new elements for each item. This recycling process is managed internally.
  
-During virtual scrolling, the data retrieved from the data source depends on the popup height and the calculation of the list item height. Enabling the [enableVirtualization](../api/drop-down-list/#enableVirtualization) option in a dropdown list activates this virtualization technique.
+During virtual scrolling, the data retrieved from the data source depends on the popup height and the calculation of the list item height. Enabling the [enableVirtualization](../api/drop-down-list/#enableVirtualization) option in a dropdown list activates this virtualization technique. take and skip queries handled based
```
Comment (PrinceOliver) [inline ej2-javascript/drop-down-list/virtual-scroll.md:16]: incomplete sentence at the end

File: ej2-javascript/drop-down-list/virtual-scroll.md [hunk 6]
```diff
@@ -13,10 +13,22 @@ domainurl: ##DomainURL##
...
  
-During virtual scrolling, the data retrieved from the data source depends on the popup height and the calculation of the list item height. Enabling the [enableVirtualization](../api/drop-down-list/#enableVirtualization) option in a dropdown list activates this virtualization technique.
+During virtual scrolling, the data retrieved from the data source depends on the popup height and the calculation of the list item height. Enabling the [enableVirtualization](../api/drop-down-list/#enableVirtualization) option in a dropdown list activates this virtualization technique. take and skip queries handled based 
  
 When fetching data from the data source, the [actionBegin](../api/drop-down-list/#actionbegin) event is triggered before data retrieval begins. Then, the [actionComplete](../api/drop-down-list/#actioncomplete) event is triggered once the data is successfully fetched.
 
+Furthermore, IncrementalSearch is supported with virtualization in the DropDownList component. When a key is typed, the focus is moved to the respective element in the open popup state. In the closed popup state, the focus shifts to the corresponding item based on the typed key.
```
Comment (PrinceOliver) [inline ej2-javascript/drop-down-list/virtual-scroll.md:20]: In the closed popup state, the focus shifts to the corresponding item based on the typed key. - What do you mean in this sentence ? 

matched items is updated in the input alone. Mentioned content is not the behavior

File: ej2-javascript/drop-down-list/virtual-scroll.md [hunk 7]
```diff
@@ -13,10 +13,22 @@ domainurl: ##DomainURL##
...
+
+Please note that when the enableVirtualization property is enabled, the 'take' property of the Query parameter may not function as expected, as it is internally managed within our source code.
```
Comment (PrinceOliver) [inline ej2-javascript/drop-down-list/virtual-scroll.md:22]: Skip property is missing here

File: ej2-javascript/drop-down-list/virtual-scroll.md [hunk 8]
```diff
@@ -13,10 +13,22 @@ domainurl: ##DomainURL##
...
+
+## Binding local data
+
+The DropDownList can generate its list items through an array of complex data. For this, the appropriate columns should be mapped to the [fields](../api/drop-down-list/#fields) property. During virtual scrolling, the list is refreshed in accordance with the scroll offset value, prompting a request to retrieve data from the server. Concurrently, as data is being sourced, the `actionBegin` event is initiated prior to the commencement of data acquisition. Following the successful retrieval of data, the `actionComplete` event is activated, marking the completion of the data fetch process.
+
+Furthermore, IncrementalSearch is supported with virtualization in the DropDownList component. When a key is typed, the focus is moved to the respective element in the open popup state. In the closed popup state, the focus shifts to the corresponding item based on the typed key. The Incremental Search functionality is well-suited for scenarios that involve remote data binding case.
```
Comment (PrinceOliver) [inline ej2-javascript/drop-down-list/virtual-scroll.md:28]: Content is repeated again

File: ej2-javascript/drop-down-list/virtual-scroll.md [hunk 9]
```diff
@@ -13,10 +13,22 @@ domainurl: ##DomainURL##
...
 
+Furthermore, IncrementalSearch is supported with virtualization in the DropDownList component. When a key is typed, the focus is moved to the respective element in the open popup state. In the closed popup state, the focus shifts to the corresponding item based on the typed key.
+
+Please note that when the enableVirtualization property is enabled, the 'take' property of the Query parameter may not function as expected, as it is internally managed within our source code.
+
+## Binding local data
+
+The DropDownList can generate its list items through an array of complex data. For this, the appropriate columns should be mapped to the [fields](../api/drop-down-list/#fields) property. During virtual scrolling, the list is refreshed in accordance with the scroll offset value, prompting a request to retrieve data from the server. Concurrently, as data is being sourced, the `actionBegin` event is initiated prior to the commencement of data acquisition. Following the successful retrieval of data, the `actionComplete` event is activated, marking the completion of the data fetch process.
```
Comment (PrinceOliver) [inline ej2-javascript/drop-down-list/virtual-scroll.md:26]: use simpler vocabulary

File: ej2-javascript/auto-complete/virtual-scroll.md [hunk 10]
```diff
@@ -17,6 +17,14 @@ During virtual scrolling, the data retrieved from the data source depends on the
  
 When fetching data from the data source, the [actionBegin](../api/auto-complete/#actionbegin) event is triggered before data retrieval begins. Then, the [actionComplete](../api/auto-complete/#actioncomplete) event is triggered once the data is successfully fetched.
```
Comment (saravanangn) [inline ej2-javascript/auto-complete/virtual-scroll.md:18]: is the action begin and action complete event triggered local data?

File: ej2-javascript/auto-complete/virtual-scroll.md [hunk 11]
```diff
@@ -17,6 +17,14 @@ During virtual scrolling, the data retrieved from the data source depends on the
...
+
+## Binding local data
+
+The AutoComplete can generate its list items through an array of complex data. For this, the appropriate columns should be mapped to the [fields](../api/drop-down-list/#fields) property. When using virtual scrolling, the list updates based on the scroll offset value, triggering a request to fetch more data from the server. As the data is being fetched, the `actionBegin` event occurs before the data retrieval starts. Once the data retrieval is successful, the `actionComplete` event is triggered, indicating that the data fetch process is complete.
+
+In the following example, `id` column and `text` column from complex data have been mapped to the `value` field and `text` field, respectively.
```
Comment (saravanangn) [inline ej2-javascript/auto-complete/virtual-scroll.md:26]: Autocomplete we should map only value field

**Review by PrinceOliver — APPROVED** [review]

**Pull request merged**